Mathlib/RingTheory/MvPolynomial/NewtonIdentities.lean

```diff
@@ -0,0 +1,271 @@
+/-
+Copyright (c) 2023 Michael Lee. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Michael Lee
+-/
+import Mathlib.Algebra.Algebra.Subalgebra.Basic
+import Mathlib.Data.Finset.Card
+import Mathlib.Data.Fintype.Basic
+import Mathlib.Data.MvPolynomial.CommRing
+import Mathlib.Data.MvPolynomial.Rename
+import Mathlib.Data.Nat.Parity
+import Mathlib.RingTheory.MvPolynomial.Basic
+import Mathlib.RingTheory.MvPolynomial.Symmetric
+import Mathlib.RingTheory.Polynomial.Basic
+
+/-!
+# Newton's Identities
+
+This file defines `MvPolynomial` power sums as a means of implementing Newton's identities. The
+combinatorial proof, due to Zeilberger, defines for `k : ℕ` a subset `pairs` of
+`(range k).powerset × range k` and a map `pairMap` such that `pairMap` is an involution on `pairs`,
+and a bijection `weight` from `pairs` to the terms of the summation in Newton's identities which
+satisfies `weight ∘ pairMap = -weight`. The result therefore follows neatly from an identity
+implemented in mathlib as `Finset.sum_involution`.
+
+## Main declarations
+
+* `MvPolynomial.mul_esymm_eq_sum`: a recurrence relation for the kth elementary
+  symmetric polynomial in terms of lower degree elementary symmetric polynomials and power sums.
+
```

Comment:
I just noticed that this argument relies on `eq_neg_self_iff` so it doesn't quite go through for all commutative rings, which is a shame. I think we should at least capture this in the doc string:
```suggestion
## TODO

Remove the `[NoZeroDivisors R] [CharZero R]` hypotheses

```

Reply:
I removed the hypotheses by adding the suggested `pairMap_ne_self` lemma, so this is actually unnecessary!